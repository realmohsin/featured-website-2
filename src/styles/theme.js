import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
