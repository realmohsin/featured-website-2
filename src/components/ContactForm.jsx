import React, { useState } from "react";
import { StaticImage } from "gatsby-plugin-image";
import { makeStyles } from "@material-ui/core/styles";
import { Container } from "@material-ui/core";
import Input from "./Input";
import Button from "./Button";

const useStyles = makeStyles((theme) => ({
  contactSection: {
    padding: "12rem 0",
    background: `radial-gradient(circle, rgba(239,249,249, 0.9) 0%, rgba(255, 255, 255, 0.4) 100%)`,
    position: "relative",
    [theme.breakpoints.down("xs")]: {
      padding: "8rem 0",
    },
  },
  contactTitle: {
    ...theme.custom.title2,
    marginBottom: "2rem",
    [theme.breakpoints.down("xs")]: {
      fontSize: "3.4rem",
    },
  },
  form: {
    maxWidth: "40rem",
    margin: "0 auto",
  },
  submitButton: {
    margin: "4rem auto",
  },
  telephoneContainer: {
    margin: "0 auto",
    width: "max-content",
    marginBottom: "8rem",
    [theme.breakpoints.down("xs")]: {
      width: "auto",
      textAlign: "center",
    },
  },
  telephoneLink: {
    color: theme.palette.secondary.dark,
    display: "inline",
    margin: "0 0.5rem",
    fontSize: "1.6rem",
    letterSpacing: "0.4px",
    transition: "0.3s all",
    "&:hover": {
      color: theme.palette.primary.main,
    },
  },
  lightBg: {
    position: "absolute !important",
    top: 0,
    bottom: 0,
    right: 0,
    left: 0,
    zIndex: -10,
    "& img": {
      objectFit: "cover",
    },
  },
}));

const ContactSection = (props) => {
  const classes = useStyles();

  const initialFormState = {
    email: "",
    firstName: "",
    lastName: "",
    telephone: "",
    message: "",
  };
  const [formState, setFormState] = useState(initialFormState);

  const handleInputChange = (e) => {
    const { value, name } = e.target;
    setFormState((formState) => ({ ...formState, [name]: value }));
  };

  return (
    <section className={classes.contactSection}>
      <StaticImage
        src="../assets/images/light-bg.png"
        className={classes.lightBg}
        alt="background"
      />
      <Container>
        <h2 className={classes.contactTitle}>How Can We Help?</h2>
        <div className={classes.telephoneContainer}>
          Call us at
          <a href="tel:123-456-7890" className={classes.telephoneLink}>
            (123) 456-7890
          </a>
          or submit a message.
        </div>

        <form data-testid="contact-form" className={classes.form}>
          <Input
            type="text"
            name="firstName"
            value={formState.firstName}
            label="First Name"
            onChange={handleInputChange}
          />
          <Input
            type="text"
            name="lastName"
            value={formState.lastName}
            label="Last Name"
            onChange={handleInputChange}
          />
          <Input
            type="tel"
            name="telephone"
            value={formState.telephone}
            label="Telephone"
            onChange={handleInputChange}
          />
          <Input
            type="email"
            name="email"
            value={formState.email}
            label="Email"
            onChange={handleInputChange}
          />
          <Input
            isTextarea
            rows={5}
            name="message"
            value={formState.message}
            label="Message"
            onChange={handleInputChange}
          />
          <Button text="Submit" extraClassName={classes.submitButton} />
        </form>
      </Container>
    </section>
  );
};

export default ContactSection;
