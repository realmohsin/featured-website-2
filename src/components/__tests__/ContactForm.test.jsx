import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import ContactForm from "../ContactForm";

// Mock Gatsby's StaticImage component
jest.mock("gatsby-plugin-image", () => ({
  StaticImage: ({ src, className, alt, ...props }) => (
    <img src={src} className={className} alt={alt} {...props} />
  ),
}));

// Mock Input and Button components
jest.mock("../Input", () => {
  return function MockInput({
    label,
    value,
    onChange,
    name,
    type,
    isTextarea,
    ...props
  }) {
    const Component = isTextarea ? "textarea" : "input";
    return (
      <div data-testid={`input-${name}`}>
        <Component
          type={type}
          name={name}
          value={value}
          onChange={onChange}
          placeholder={label}
          {...props}
        />
        {label && <label>{label}</label>}
      </div>
    );
  };
});

jest.mock("../Button", () => {
  return function MockButton({ text, extraClassName, ...props }) {
    return (
      <button className={extraClassName} {...props}>
        {text}
      </button>
    );
  };
});

const theme = createMuiTheme({
  palette: {
    primary: { main: "#0E5C60", light: "#4A8B8F" },
    secondary: { main: "#CCAB63", dark: "#B8965A" },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
  custom: {
    title2: {
      fontSize: "4.8rem",
      fontWeight: "bold",
    },
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("ContactForm Component", () => {
  it("renders contact form with all elements", () => {
    const { container } = renderWithTheme(<ContactForm />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("displays contact title and phone information", () => {
    renderWithTheme(<ContactForm />);
    expect(screen.getByText(/How Can We Help?/i)).toBeInTheDocument();
    expect(screen.getByText(/call us at/i)).toBeInTheDocument();
    expect(screen.getByText(/\(123\) 456-7890/)).toBeInTheDocument();
    expect(screen.getByText(/or submit a message\./i)).toBeInTheDocument();
  });

  it("renders phone link with correct href", () => {
    renderWithTheme(<ContactForm />);

    const phoneLink = screen.getByRole("link", { name: "(123) 456-7890" });
    expect(phoneLink).toHaveAttribute("href", "tel:123-456-7890");
  });

  it("renders all form inputs", () => {
    renderWithTheme(<ContactForm />);

    expect(screen.getByTestId("input-firstName")).toBeInTheDocument();
    expect(screen.getByTestId("input-lastName")).toBeInTheDocument();
    expect(screen.getByTestId("input-telephone")).toBeInTheDocument();
    expect(screen.getByTestId("input-email")).toBeInTheDocument();
    expect(screen.getByTestId("input-message")).toBeInTheDocument();
  });

  it("renders submit button", () => {
    renderWithTheme(<ContactForm />);

    const submitButton = screen.getByRole("button", { name: "Submit" });
    expect(submitButton).toBeInTheDocument();
  });

  it("updates form state when input values change", () => {
    renderWithTheme(<ContactForm />);

    const firstNameInput = screen
      .getByTestId("input-firstName")
      .querySelector("input");
    const lastNameInput = screen
      .getByTestId("input-lastName")
      .querySelector("input");
    const emailInput = screen.getByTestId("input-email").querySelector("input");

    fireEvent.change(firstNameInput, { target: { value: "John" } });
    fireEvent.change(lastNameInput, { target: { value: "Doe" } });
    fireEvent.change(emailInput, { target: { value: "john@example.com" } });

    expect(firstNameInput.value).toBe("John");
    expect(lastNameInput.value).toBe("Doe");
    expect(emailInput.value).toBe("john@example.com");
  });

  it("updates textarea value", () => {
    renderWithTheme(<ContactForm />);

    const messageTextarea = screen
      .getByTestId("input-message")
      .querySelector("textarea");

    fireEvent.change(messageTextarea, {
      target: { value: "This is a test message for the contact form." },
    });

    expect(messageTextarea.value).toBe(
      "This is a test message for the contact form."
    );
  });

  it("renders background image", () => {
    renderWithTheme(<ContactForm />);

    const backgroundImage = screen.getByAltText("background");
    expect(backgroundImage).toBeInTheDocument();
    expect(backgroundImage).toHaveAttribute(
      "src",
      "../assets/images/light-bg.png"
    );
  });

  it("renders with correct input types", () => {
    renderWithTheme(<ContactForm />);

    const firstNameInput = screen
      .getByTestId("input-firstName")
      .querySelector("input");
    const telephoneInput = screen
      .getByTestId("input-telephone")
      .querySelector("input");
    const emailInput = screen.getByTestId("input-email").querySelector("input");

    expect(firstNameInput).toHaveAttribute("type", "text");
    expect(telephoneInput).toHaveAttribute("type", "tel");
    expect(emailInput).toHaveAttribute("type", "email");
  });
});
