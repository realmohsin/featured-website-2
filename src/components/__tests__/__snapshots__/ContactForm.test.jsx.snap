// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm Component renders contact form with all elements 1`] = `
<section
  class="makeStyles-contactSection-1"
>
  <img
    alt="background"
    class="makeStyles-lightBg-7"
    src="../assets/images/light-bg.png"
  />
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h2
      class="makeStyles-contactTitle-2"
    >
      How Can We Help?
    </h2>
    <div
      class="makeStyles-telephoneContainer-5"
    >
      Call us at
      <a
        class="makeStyles-telephoneLink-6"
        href="tel:123-456-7890"
      >
        (123) 456-7890
      </a>
      or submit a message.
    </div>
    <form
      class="makeStyles-form-3"
      data-testid="contact-form"
    >
      <div
        data-testid="input-firstName"
      >
        <input
          name="firstName"
          placeholder="First Name"
          type="text"
          value=""
        />
        <label>
          First Name
        </label>
      </div>
      <div
        data-testid="input-lastName"
      >
        <input
          name="lastName"
          placeholder="Last Name"
          type="text"
          value=""
        />
        <label>
          Last Name
        </label>
      </div>
      <div
        data-testid="input-telephone"
      >
        <input
          name="telephone"
          placeholder="Telephone"
          type="tel"
          value=""
        />
        <label>
          Telephone
        </label>
      </div>
      <div
        data-testid="input-email"
      >
        <input
          name="email"
          placeholder="Email"
          type="email"
          value=""
        />
        <label>
          Email
        </label>
      </div>
      <div
        data-testid="input-message"
      >
        <textarea
          name="message"
          placeholder="Message"
          rows="5"
        />
        <label>
          Message
        </label>
      </div>
      <button
        class="makeStyles-submitButton-4"
      >
        Submit
      </button>
    </form>
  </div>
</section>
`;
